packages/framework-integration-tests/integration/providers/aws/utils.ts

```diff
@@ -554,3 +366,70 @@ export async function countSnapshotItems(entityTypeName: string, entityID: strin
   return output.Count ?? -1
 }
 
+export async function getEventsByEntityId(entityID: string): Promise<any> {
+  const output: QueryOutput = await documentClient
+    .scan({
+      TableName: await eventsStoreTableName(),
+      Select: 'ALL_ATTRIBUTES',
+      ExpressionAttributeNames: { '#entityID': 'entityID' },
+      ExpressionAttributeValues: { ':entityID': entityID },
+      FilterExpression: '#entityID = :entityID',
+    })
+    .promise()
+
+  return output.Items
+}
+
+// --- Other helpers ---
+
+export async function waitForIt<TResult>(
+  tryFunction: () => Promise<TResult>,
+  checkResult: (result: TResult) => boolean,
+  trialDelayMs = 1000,
+  timeoutMs = 60000
+): Promise<TResult> {
+  console.debug('[waitForIt] start')
+  const start = Date.now()
+  return doWaitFor()
+
+  async function doWaitFor(): Promise<TResult> {
+    console.debug('.')
+    const res = await tryFunction()
+    if (checkResult(res)) {
+      console.debug('[waitForIt] match!')
+      return res
+    }
+    const elapsed = Date.now() - start
+
+    if (elapsed > timeoutMs) {
+      throw new Error('[waitForIt] Timeout reached')
+    }
+
+    await sleep(trialDelayMs)
+    return doWaitFor()
+  }
+}
+
+// This helper will create a valid token using a real private key for testing
+// the tokens will be validate against the public keyset file
+// located in: /keys/private.key file
+export const getTokenForUser = (email: string, role: string): string => {
```

Comment:
This method is the key to perform the integration tests using valid tokens. It will create token, signed with a private key, containing the user envelop including the role.